repose-aggregator/components/filters/keystone-v2/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -0,0 +1,360 @@
+ */
+package org.openrepose.filters.keystonev2
+
+import java.net.URL
+import javax.inject.{Inject, Named}
+import javax.servlet._
+import javax.servlet.http.HttpServletResponse._
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+
+import com.rackspace.httpdelegation.HttpDelegationManager
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.apache.commons.codec.binary.Base64
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.http._
+import org.openrepose.commons.utils.servlet.http.MutableHttpServletRequest
+import org.openrepose.core.filter.FilterConfigHelper
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.core.services.datastore.{Datastore, DatastoreService}
+import org.openrepose.core.services.serviceclient.akka.AkkaServiceClient
+import org.openrepose.core.systemmodel.SystemModel
+import org.openrepose.filters.keystonev2.RequestHandler._
+import org.openrepose.filters.keystonev2.config._
+
+import scala.util.{Failure, Success, Try}
+
+@Named
+class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
+                                 akkaServiceClient: AkkaServiceClient,
+                                 datastoreService: DatastoreService)
+  extends Filter
+  with HttpDelegationManager
+  with LazyLogging {
+
+  private final val SYSTEM_MODEL_CONFIG = "system-model.cfg.xml"
+  private final val DEFAULT_CONFIG = "keystone-v2.cfg.xml"
+  private final val X_AUTH_PROXY = "Proxy"
+
+  private var configurationFile: String = DEFAULT_CONFIG
+  private var sendTraceHeader = true
+
+  //Which happens to be the local datastore
+  private val datastore: Datastore = datastoreService.getDefaultDatastore
+
+  var keystoneV2Config: KeystoneV2Config = _
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    configurationFile = new FilterConfigHelper(filterConfig).getFilterConfig(DEFAULT_CONFIG)
+    logger.info(s"Initializing Keystone V2 Filter using config $configurationFile")
+    val xsdURL: URL = getClass.getResource("/META-INF/schema/config/keystone-v2.xsd")
+    configurationService.subscribeTo(
+      SYSTEM_MODEL_CONFIG,
+      getClass.getResource("/META-INF/schema/system-model/system-model.xsd"),
+      SystemModelConfigListener,
+      classOf[SystemModel]
+    )
+    configurationService.subscribeTo(
+      filterConfig.getFilterName,
+      configurationFile,
+      xsdURL,
+      KeystoneV2ConfigListener,
+      classOf[KeystoneV2Config]
+    )
+  }
+
+  override def destroy(): Unit = {
+    configurationService.unsubscribeFrom(configurationFile, KeystoneV2ConfigListener)
+    configurationService.unsubscribeFrom(SYSTEM_MODEL_CONFIG, SystemModelConfigListener)
+  }
+
+  implicit val autoHeaderToString: HeaderConstant => String = hc => hc.toString
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, chain: FilterChain): Unit = {
```

Comment:
this method is way to long, it should be broken up into a number of smaller well named methods